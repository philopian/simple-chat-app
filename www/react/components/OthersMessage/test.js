import React from "react";
import Enzyme, { configure, shallow, mount, render } from "enzyme";
import renderer from "react-test-renderer";
import "raf/polyfill";
import Adapter from "enzyme-adapter-react-16";
import Component from "./index.js";

Enzyme.configure({ adapter: new Adapter() });

describe("OthersMessage (Snapshot)", () => {
  it("OthersMessage renders without crashing", () => {
    const component = renderer.create(<Component />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

it("Make sure the 'username' props are being passing thru to the component", () => {
  // Render the component
  const SOME_PROP = "some username prop";
  const component = shallow(<Component username={SOME_PROP} />);

  // Make sure that the props.username is being printed
  expect(component.find(".username").text()).toEqual(SOME_PROP);
});

it("Make sure the 'message' props are being passing thru to the component", () => {
  // Render the component
  const SOME_PROP = "some message prop";
  const component = shallow(<Component message={SOME_PROP} />);

  // Make sure that the props.message is being printed
  expect(component.find(".message").text()).toEqual(SOME_PROP);
});

it("Make sure the 'timestamp' props are being passing thru to the component", () => {
  // Render the component
  const SOME_PROP = "some timestamp prop";
  const component = shallow(<Component timestamp={SOME_PROP} />);

  // Make sure that the props.timestamp is being printed
  expect(component.find(".timestamp").text()).toEqual(SOME_PROP);
});

it("Make sure that all props are being passing thru to the component", () => {
  // Render the component
  const SOME_PROP = {
    username: "some username",
    message: "some message",
    timestamp: "some timestamp"
  };
  const component = shallow(<Component {...SOME_PROP} />);

  // Make sure that the props.timestamp is being printed
  expect(component.find(".timestamp").text()).toEqual(SOME_PROP.timestamp);
  expect(component.find(".message").text()).toEqual(SOME_PROP.message);
  expect(component.find(".timestamp").text()).toEqual(SOME_PROP.timestamp);
});
