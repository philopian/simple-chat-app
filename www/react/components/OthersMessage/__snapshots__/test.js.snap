// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OthersMessage (Snapshot) OthersMessage renders without crashing 1`] = `
<div
  className="sc-bdVaJa gAtwGl"
>
  <div
    className="arrow-left"
  />
  <div
    className="content"
  >
    <div
      className="username"
    >
      
    </div>
    <div
      className="message"
    >
      
    </div>
    <div
      className="timestamp"
    >
      
    </div>
    <div
      className="deal-with-float"
    />
  </div>
</div>
`;
